spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/ribbon/apache/HttpClientRibbonConfiguration.java

```diff
@@ -81,21 +81,22 @@ public HttpClientConnectionManager httpClientConnectionManager(
 			Integer timerRepeat = config.getPropertyAsInteger(
 					CommonClientConfigKey.ConnectionCleanerRepeatInterval,
 					DefaultClientConfigImpl.DEFAULT_CONNECTION_IDLE_TIMERTASK_REPEAT_IN_MSECS);
-			Object timeToLiveObj = config
-					.getProperty(CommonClientConfigKey.PoolKeepAliveTime);
-			Long timeToLive = DefaultClientConfigImpl.DEFAULT_POOL_KEEP_ALIVE_TIME;
+			// Since we cannot change the type of CommonClientConfigKey.PoolKeepAliveTime from Integer to Long,
+			// we are working with integers.
+			// Unfortunately, the default is defined as long in the same package whereas the config key is using
+			// Integer.
+			Integer timeToLive = config.get(
+					CommonClientConfigKey.PoolKeepAliveTime,
+					(int)DefaultClientConfigImpl.DEFAULT_POOL_KEEP_ALIVE_TIME);
 			Object ttlUnitObj = config
 					.getProperty(CommonClientConfigKey.PoolKeepAliveTimeUnits);
 			TimeUnit ttlUnit = DefaultClientConfigImpl.DEFAULT_POOL_KEEP_ALIVE_TIME_UNITS;
-			if (timeToLiveObj instanceof Long) {
```

Comment:
Can we just check if it is a String and then convert the string to a Long?

Reply:
Sure.